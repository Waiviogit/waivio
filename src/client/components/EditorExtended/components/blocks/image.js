import PropTypes from 'prop-types';
import React from 'react';

import { EditorBlock, EditorState, SelectionState } from 'draft-js';

import { getCurrentBlock } from '../../model';
import './image.less';

class ImageBlock extends React.Component {
  static propTypes = {
    block: PropTypes.shape().isRequired,
    blockProps: PropTypes.shape(),
  };
  static defaultProps = {
    blockProps: {},
  };

  focusBlock = () => {
    const { block, blockProps } = this.props;
    const { getEditorState, setEditorState } = blockProps;
    const key = block.getKey();
    const editorState = getEditorState();
    const currentBlock = getCurrentBlock(editorState);

    if (currentBlock.getKey() === key) {
      return;
    }
    const newSelection = new SelectionState({
      anchorKey: key,
      focusKey: key,
      anchorOffset: 0,
      focusOffset: 0,
    });

    setEditorState(EditorState.forceSelection(editorState, newSelection));
  };

  render() {
    const { block } = this.props;
    const data = block.getData();
    const src = data.get('src');

    if (src !== null) {
      return (
        <div
          role="presentation"
          className="md-block-image-inner-container"
          onClick={this.focusBlock}
        >
          <img alt={data.get('alt') || ''} src={src} />
        </div>
      );
    }

    return <EditorBlock {...this.props} />;
  }
}

export default ImageBlock;
