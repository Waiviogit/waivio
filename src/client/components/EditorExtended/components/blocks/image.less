figure.md-block-image {
  img {
    cursor: default;

    &.is-selected {
      box-shadow: 0 0 0 3px #02b875;
    }
  }

  figcaption {
    padding: 25px 0;

    .public-DraftStyleDefault-block {
      text-align: center;
    }
  }

  .md-block-image-inner-container {
    position: relative;
    text-align: center;
  }

  .md-block-image-toolbar-container {
    position: absolute;
    top: 0;
    right: 10px;
    cursor: pointer;
  }
}
