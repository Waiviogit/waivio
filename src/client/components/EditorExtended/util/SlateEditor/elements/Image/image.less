.element-image {
  width: 100%;
}

.element-image img {
  width: 100%;
  height: 100%;
}

.md-block-image-figure {
  margin: 16px 0;
  text-align: center;
}

.md-block-image-caption {
  font-size: 15px;
  font-style: italic;
  text-align: center;
  margin-top: 8px;
  color: #666;
  line-height: 1.4;
}
