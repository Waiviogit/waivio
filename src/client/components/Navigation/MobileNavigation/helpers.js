import { CAMPAIGNS } from '../../../../common/constants/rewards';

export const pages = {
  hive: {
    trending: 'trending',
    regExp: /(^\/)(created|hot|trending)$/,
    id: 'hive',
  },
  personal: {
    myFeed: 'my_feed',
    regExp: /(^\/)(notifications-list|updates|rewards-list)$/,
    id: 'personal',
  },
  people: {
    regExp: /(^\/blog)\/(@[\w\d.-]{3,})$/,
    id: 'people',
  },
  objectsUpdates: {
    regExp: /(^\/feed\/)([\w\d.-]{3,})/,
  },
  discoverObjects: {
    all: 'all',
    regExp: /(^\/discover-objects)\/?(.*)/,
    id: 'objects',
  },
  rewards: {
    regExp: /(^\/rewards\/)(all|active|reserved|receivables|history)/,
    id: 'rewards',
  },
  rewardsCampaigns: {
    regExp: /(^\/rewards\/)(create|manage|payables|reservations|messages|match-bot|blacklist)$/,
    id: CAMPAIGNS,
  },
};

export default {};
