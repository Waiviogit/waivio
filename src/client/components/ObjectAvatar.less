@import (reference) '../styles/custom.less';

.ObjectAvatar {
  display: block !important;
  //border-radius: 50%;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
  border: @base-border;
  background-color: @white;
  cursor: pointer;
}
