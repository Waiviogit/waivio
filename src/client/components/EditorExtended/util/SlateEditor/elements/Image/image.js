import React, { useCallback } from 'react';
import { Transforms } from 'slate';
import { useSelected, useFocused, ReactEditor, useSlateStatic } from 'slate-react';
import PropTypes from 'prop-types';
import { shouldShowCaption } from '../../../../../../../common/helpers/imageCaption';

import './image.less';

const Image = ({ attributes, element, children }) => {
  const selected = useSelected();
  const focused = useFocused();
  const editor = useSlateStatic();

  const selectImage = useCallback(
    e => {
      e.preventDefault();
      const path = ReactEditor.findPath(editor, element);

      Transforms.select(editor, path);
      ReactEditor.focus(editor);
    },
    [editor, element],
  );

  return (
    <div
      {...attributes}
      className="md-block-image-inner-container"
      role="presentation"
      onMouseDown={selectImage}
      style={{ textAlign: 'center' }}
    >
      <div contentEditable={false}>
        <figure className="md-block-image-figure">
          <img
            alt={element.alt || ''}
            src={element.url}
            data-fallback-src={element.url}
            draggable={false}
            style={{
              boxShadow: selected && focused && '0 0 3px 3px lightgray',
              textAlign: 'center',
              display: 'inline-block',
              marginTop: 0,
            }}
          />
          {shouldShowCaption(element.alt, element.url) ? (
            <figcaption className="md-block-image-caption">{element.alt}</figcaption>
          ) : null}
        </figure>
      </div>
      {children}
    </div>
  );
};

Image.propTypes = {
  attributes: PropTypes.shape().isRequired,
  element: PropTypes.shape().isRequired,
  children: PropTypes.node.isRequired,
};

export default Image;
