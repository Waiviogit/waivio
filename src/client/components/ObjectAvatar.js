import React from 'react';
import PropTypes from 'prop-types';
import { includes, get } from 'lodash';

import DEFAULTS from '../object/const/defaultValues';
import { getProxyImageURL } from '../helpers/image';

import './ObjectAvatar.less';

const ObjectAvatar = ({ item, size }) => {
  let style = {
    minWidth: `${size}px`,
    width: `${size}px`,
    height: `${size}px`,
  };
  const parent = get(item, ['parent'], {});
  let url = item.avatar || parent.avatar;

  if (url) url = getProxyImageURL(url, 'preview');
  else url = DEFAULTS.AVATAR;

  if (includes(url, 'waivio.')) url = `${url}${size < 41 ? '_small' : '_medium'}`;

  style = {
    ...style,
    backgroundImage: `url(${url})`,
  };

  return <div className="ObjectAvatar" style={style} />;
};

ObjectAvatar.propTypes = {
  item: PropTypes.shape({
    parent: PropTypes.oneOfType([PropTypes.shape(), PropTypes.string]),
    avatar: PropTypes.string,
  }),
  size: PropTypes.number,
};

ObjectAvatar.defaultProps = {
  size: 100,
  item: {},
};

export default ObjectAvatar;
