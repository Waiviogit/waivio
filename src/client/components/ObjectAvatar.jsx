import React from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash';

import DEFAULTS from '../object/const/defaultValues';
import { getProxyImageURL } from '../../common/helpers/image';

import './ObjectAvatar.less';

const ObjectAvatar = ({ item, size, className }) => {
  let style = {
    minWidth: `${size}px`,
    width: `${size}px`,
    height: `${size}px`,
  };
  const parent = get(item, ['parent'], {});
  let url = item.avatar || parent.avatar;

  if (url) url = getProxyImageURL(url, 'preview');
  else url = DEFAULTS.AVATAR;

  style = {
    ...style,
    backgroundImage: `url(${url})`,
  };

  return <div className={`ObjectAvatar ${className}`} style={style} />;
};

ObjectAvatar.propTypes = {
  item: PropTypes.shape({
    parent: PropTypes.oneOfType([PropTypes.shape(), PropTypes.string]),
    avatar: PropTypes.string,
  }),
  size: PropTypes.number,
  className: PropTypes.string,
};

ObjectAvatar.defaultProps = {
  size: 100,
  item: {},
  className: '',
};

export default ObjectAvatar;
