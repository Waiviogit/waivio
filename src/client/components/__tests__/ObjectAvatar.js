import React from 'react';
import { shallow } from 'enzyme';
import DEFAULTS from '../../object/const/defaultValues';
import ObjectAvatar from '../ObjectAvatar';
import { getProxyImageURL } from '../../helpers/image';

describe('<ObjectAvatar />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      item: {
        parent: { avatar: 'parent_avatar' },
        avatar: 'avatar',
      },
      size: 42,
    };

    wrapper = shallow(<ObjectAvatar {...props} />);
  });

  afterEach(() => jest.clearAllMocks());

  it('renders without exploding', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be item avatar', () => {
    expect(wrapper.props().style.backgroundImage).toBe(
      `url(${getProxyImageURL(props.item.avatar, 'preview')})`,
    );
    expect(wrapper.props().style.width).toBe(`${props.size}px`);
    expect(wrapper.props().style.height).toBe(`${props.size}px`);
  });

  it('have to get parent avatar, if don`t have item avatar', () => {
    const currProps = {
      avatar: '',
      item: {
        parent: {
          avatar: 'parentAvatar',
        },
      },
    };
    wrapper = shallow(<ObjectAvatar {...currProps} />);

    expect(wrapper.props().style.backgroundImage).toBe(
      `url(${getProxyImageURL(currProps.item.parent.avatar, 'preview')})`,
    );
  });

  it('should have default url with prefix _medium', () => {
    const currProps = {
      item: {},
      size: 42,
    };
    wrapper = shallow(<ObjectAvatar {...currProps} />);

    expect(wrapper.props().style.backgroundImage).toBe(`url(${DEFAULTS.AVATAR}_medium)`);
  });

  it('should have default url with prefix _small', () => {
    const currProps = {
      item: {},
      size: 39,
    };
    wrapper = shallow(<ObjectAvatar {...currProps} />);

    expect(wrapper.props().style.backgroundImage).toBe(`url(${DEFAULTS.AVATAR}_small)`);
  });

  it('should have style rules as prop size', () => {
    props.item = {};
    const style = {
      minWidth: `${props.size}px`,
      width: `${props.size}px`,
      height: `${props.size}px`,
      backgroundImage: `url(${DEFAULTS.AVATAR}_medium)`,
    };
    wrapper = shallow(<ObjectAvatar {...props} />);

    expect(wrapper.props().style).toEqual(style);
  });
});
