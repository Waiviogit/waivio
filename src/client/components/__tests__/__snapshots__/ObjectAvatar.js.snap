// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ObjectAvatar /> renders without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ObjectAvatar
    item={
      Object {
        "avatar": "avatar",
        "parent": Object {
          "avatar": "parent_avatar",
        },
      }
    }
    size={42}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "ObjectAvatar",
      "style": Object {
        "backgroundImage": "url(https://images.hive.blog/800x600/https://images.hive.blog/p/qXwViYfK)",
        "height": "42px",
        "minWidth": "42px",
        "width": "42px",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "ObjectAvatar",
        "style": Object {
          "backgroundImage": "url(https://images.hive.blog/800x600/https://images.hive.blog/p/qXwViYfK)",
          "height": "42px",
          "minWidth": "42px",
          "width": "42px",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
